Automattic/wp-calypso
# Add correct label to back button override

### Goal:
1. Show the correct back button label in the FSE editor.
2. Conveniently, this gave us an opportunity to improve how we set the back button URL and label using a nifty action from `wp.hooks`.

<img width="961" alt="Screen Shot 2019-11-19 at 9 21 45 PM" src="https://user-images.githubusercontent.com/6265975/69211343-a728b080-0b12-11ea-8a43-33e2efedfe8c.png">

### Changes proposed in this Pull Request
* Remove explicit selector for the checklist case and use the generic one introduced in #36833
* Add support for going back to the "themes" page since we can navigate from it to the editor
* Add a label to each custom back Url
* Refactor the button override so that it can be more reactive to updates of the back button URL/label.
* Pass the label into the iframe bridge server, which assigns it to the FSE global object.

TBD: Load the back button file at all times regardless of if FSE is active.

#### Testing instructions
1. Pull this PR and run calypso.localhost
2. Sync the bridge server changes to the sandbox and sandbox widgets.wp.com
3. Sync the FSE changes to the sandbox and sandbox the api
3. Verify that if you go from "edit homepage" in customer home to the page editor, it shows the "Home" label on the back button, and that clicking on it goes back to home. Verify the same when you activate a theme.
4. Note that you must **not** go to the template editor, or nothing in step 3 will work. (That will be fixed by #37748)

Fixes #(tbd)
